src/WebJobs.Script.WebHost/Security/SecretsMetadata.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.Config;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class SecretsMetadata
+    {
+        public SecretsMetadata()
+        {
+            InstanceId = ScriptSettingsManager.Instance.InstanceId;
+            Source = ScriptConstants.Runtime;
+        }
+
+        [JsonProperty(PropertyName = "source")]
+        public string Source { get; set; }
+
+        [JsonProperty(PropertyName = "instanceId")]
```

Comment:
I believe a site identifier (as opposed to the instance ID) would be more helpful here, as it would allow us to more easily identify scenarios where the secrets were moved from a different site.